### DISCUSSION TITLE
why keep recalculating same thing over and over?

### DISCUSSION THREAD

**Discussion by Awersang** [ActivityWatch/activitywatch#1045] [Ideas]
Say I want to check what I was doing two months ago, and I haven't changed the categorization options. Why does the ActivityWatch recalculates the whole data for the period?
Wouldn't it be nicer if for every period the AW would keep a temporary data about categorization? In case no options have been changed since last categorization the data would be just presented (fast) and recalculated only if there is an actual need for that (slow)?

**Comment by ErikBjare** [comment]
Recent beta versions of the web UI does this, but the cache currently only persists within the loaded page iirc, so there is more work to be done here. Cache invalidation is tricky, but I have ideas for heuristics we can use.

Caching could also be done on the HTTP layer, if we use the request data as the cache key, but I haven't investigated this.
